ui/src/adapters/api/identities.ts

```diff
@@ -0,0 +1,193 @@
+import axios from "axios";
+import { z } from "zod";
+
+import { Response, buildErrorResponse, buildSuccessResponse } from "src/adapters";
+import { buildAuthorizationHeader } from "src/adapters/api";
+import { getListParser, getStrictParser } from "src/adapters/parsers";
+import {
+  CredentialStatusType,
+  Env,
+  Identifier,
+  Identity,
+  IdentityDetails,
+  IdentityType,
+  Method,
+  SupportedNetwork,
+} from "src/domain";
+
+import { API_VERSION } from "src/utils/constants";
+import { List } from "src/utils/types";
+
+const identityParser = getStrictParser<Identity>()(
+  z.object({
+    blockchain: z.string(),
+    credentialStatusType: z.nativeEnum(CredentialStatusType),
+    displayName: z.string(),
+    identifier: z.string(),
+    method: z.nativeEnum(Method),
+    network: z.string(),
+  })
+);
+
+export const identifierParser = getStrictParser<Identifier>()(z.string());
+
+export async function getIdentities({
+  env,
+  signal,
+}: {
+  env: Env;
+  signal: AbortSignal;
+}): Promise<Response<List<Identity>>> {
+  try {
+    const response = await axios({
+      baseURL: env.api.url,
+      headers: {
+        Authorization: buildAuthorizationHeader(env),
+      },
+      method: "GET",
+      signal,
+      url: `${API_VERSION}/identities`,
+    });
+
+    return buildSuccessResponse(getListParser(identityParser).parse(response.data || []));
```

Comment:
Why doing this `|| []`?

Reply:
because the request returns null when identities do not exist